lavalink/node.py

```diff
@@ -176,26 +176,28 @@ async def _multi_try_connect(self, uri):
         backoff = ExponentialBackoff()
         attempt = 1
 
-        while self._is_shutdown is False and (self._ws is None or not self._ws.open):
+        while self._is_shutdown is False and (self._ws is None or self._ws.closed):
             try:
-                ws = self._ws = await websockets.connect(uri, extra_headers=self.headers)
+                ws = self._ws = await self.session.ws_connect(url=uri, headers=self.headers)
                 return ws
             except OSError:
                 delay = backoff.delay()
                 log.debug("Failed connect attempt %s, retrying in %s", attempt, delay)
                 await asyncio.sleep(delay)
                 attempt += 1
-            except websockets.InvalidStatusCode:
+            except Exception as error:
+                log.debug(error)
                 return None
 
     async def listener(self):
         """
         Listener task for receiving ops from Lavalink.
         """
-        while self._ws.open and self._is_shutdown is False:
+        while not self._ws.closed and self._is_shutdown is False:
             try:
-                data = json.loads(await self._ws.recv())
-            except websockets.ConnectionClosed:
+                data = await self._ws.receive_json()
+            except Exception as error:
+                log.debug("listener error: ", exc_info=error)
                 break
```

Comment:
This is overgeneral exception handler, we could know exactly what message we've got by using `ws.receive()`:
```py
from aiohttp import WSMsgType

...

msg = await self._ws.receive()
if msg.type != WSMsgType.TEXT:
    if msg.type is WSMsgType.ERROR:
        log.debug("Ignoring exception in listener task", exc_info=msg.data)
    elif msg.type is WSMsgType.BINARY:
        # can that even happen? cause if not, it should probably not be on debug level
        log.debug("Received message is not str")
    # not sure if we want logging for close msg type too, but the if is below if needed
    # elif msg.type in (WSMsgType.CLOSE, WSMsgType.CLOSING, WSMsgType.CLOSED):

    break

data = json.loads(msg.data)
```